batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java

```diff
@@ -243,6 +248,26 @@ public void createBatchCustomEventTest()
                     .anyMatch(runnable -> runnable.getDisplayName().equals(displayName))));
   }
 
+  @Test
+  public void createBatchUsingNfsTest()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    Job job = CreateWithNfs.createScriptJobWithNfs(PROJECT_ID, REGION, NFS_JOB_NAME,
+            NFS_NAME, SERVER_NAME);
+
+    Assert.assertNotNull(job);
+    ACTIVE_JOBS.add(job);
+
+    Assert.assertTrue(job.getName().contains(NFS_JOB_NAME));
+
+    Assert.assertTrue(job.getTaskGroupsList().stream().anyMatch(taskGroup
+            -> taskGroup.getTaskSpec().getVolumesList().stream()
+            .anyMatch(volume -> volume.getNfs().getRemotePath().equals(NFS_NAME))));
+    Assert.assertTrue(job.getTaskGroupsList().stream().anyMatch(taskGroup
+            -> taskGroup.getTaskSpec().getVolumesList().stream()
+            .anyMatch(volume -> volume.getNfs().getServer().equals(SERVER_NAME))));
```

Comment:
Instead of iterating through all jobs here, can we retrieve this specific job and verify if the volume contains the NFS_NAME and SERVER_NAME?

Reply:
If I understand correctly, I iterate through the TaskGroupList and then the VolumesList for the specific job. Could you explain your point of view?

Reply:
On taking a second look, you are correct. I was assuming you were iterating on the jobs, but that is not the case.